### DISCUSSION TITLE
Tweaking Nominatim for specific purpose of Administrative Level Search

### DISCUSSION THREAD

**Discussion by dbt-lucka** [osm-search/Nominatim#3041] [Q&A]
Hello,

I am trying to use Nominatim to geocode phrases like 

- Hamburg-Langenhorn (a **Quarter** of Hamburg, Germany)
- Friesland (Landkreis Friesland, a **district** in Germany)
- Niedersachsen (a **state** in Germany)
- Berlin (**city**)
- Tesa, Norderstedt (A **POI** in Germany)

However, I am not exactly sure what configuration of Nominatim I have to look at and what to configure in particular. 
Additionally, I have a couple of problems with these example search terms: 

- Friesland doesn't find Landkreis Friesland
- Hamburg Langenhorn is not related to the district, only "Langenhorn, Hamburg", does.
- Tennis Hamburg is not relevant and I'd like to filter specific POI types out. 
- "Bre" should find Bremen

Are my issues even solvable with different config, i. e. will I be able to configure Nominatim in a way that it can relate "Friesland" to "Landkreis Friesland"?

I also tried https://nominatim.openstreetmap.org/search?q=Friesland&format=json&boundary=administrative 
However, I only get it returns once I hit "more result" (exclude the found ones from the result)

Ps: Is outdooractive using Nominatim? If so, they managed to get "Friesland" working. https://www.outdooractive.com/en/map

**Comment by mtmail** [comment]
That's many questions in one. Nominatim can be used for all these use cases. There can be limitations in data (OpenStreetMap) and some in Nominatim logic.

OutdoorActive uses type-ahead / search-as-you-type, that's not something Nominatim provides. They might use [Photon](https://photon.komoot.io/) which imports data from Nominatim though, hard to tell from looking at the website.

"Bre" should find Bremen: See my comment about Photon.

Friesland: Sometimes administrative places like cities or counties exist twice in OpenStreetMap data. First as boundary (https://www.openstreetmap.org/relation/62773) and as center point (https://www.openstreetmap.org/node/240068813). They can have different names even. Nominatim tries to link them together. It used to be by name and location (one inside the boundary of the other) but the wikipedia/wikidata tags are more reliable. I see OSM user lonvia added the mapping on the relation recently, possibly after reading your question. https://nominatim.openstreetmap.org/ hasn't automatically picked up the change. I tested it on my installation and after running `nominatim refresh --data-area R62773; nominatim index` (takes about one minute) it works.

Hamburg-Langenhorn: The dash is not treated as phrase boundary and treated as a single name. In Germany there's names with dashes for example. Wanne-Eickel (two cities that merged), Dortmund-Ems-Kanal for example. As you already wrote it should be written with comma.

Tesa, Norderstedt: Official company name is 'Tesa SE'. If the goal is to match company names in office buildings there will be many cases were you'd need to cleanup or change the query first.

Berlin: It's a city and and a state. When you search it gets returned as 'city' which is apparently what you expect. Other users will argue it should be returned as 'state'. It's not easy to serve both use cases.

Tennis Hamburg: You can configure Nominatim not to import certain OSM objects https://nominatim.org/release-docs/latest/customize/Import-Styles/ There's no URL parameters to filter. Photon has some URL filter parameters https://github.com/komoot/photon#filter-results-by-tags-and-values

**Reply by lonvia** [reply]
Hamburg-Langenhorn is a mapping issue, too. Somebody tagged the location into `alt_name` of [this action store](https://www.openstreetmap.org/node/490531846). This should go into `branch`. Probably needs fixing for the entire Maproulette challenge. Hard to filter out these kind of tagging errors. Sadly they have a rather huge impact on search quality.
